src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+
+    private final K[] keys = (K[]) new Object[MAX_SIZE];
+    private final V[] values = (V[]) new Object[MAX_SIZE];
+
+    private int size;
+
     @Override
     public void put(K key, V value) {
+        int index = indexOf(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size == MAX_SIZE) {
+                throw new RuntimeException("Storage is full");
+            }
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
```

Comment:
dont complicate if-else logic
```suggestion
            return;
        } 
        if (size == MAX_SIZE) {
            throw new RuntimeException("Storage is full");
        }
        keys[size] = key;
        values[size] = value;
        size++;
```